lib/requester/core.js

```diff
@@ -294,7 +295,7 @@ module.exports = {
 
         // The underlying Node client does add the host header by itself, but we add it anyway, so that
         // it is bubbled up to us after the request is made. If left to the underlying core, it's not :/
-        self.ensureHeaderExists(options.headers, 'Host', request.url.getRemote());
+        self.ensureHeaderExists(options.headers, 'Host', url.host);
```

Comment:
Add a test where the domain has Unicode chars and use `hostLookup` options (check dns-lookup tests) which resolves that host to postman-echo.

using `url.host` here fixed an issue which earlier used to fail because `request.url.getRemote()` dosn't do punycode conversion.